### PR TITLE
Added demos and si_utils rearrangement

### PR DESCRIPTION
**Pull request by NaderMansouri**
Hi, 

I added two demos using voice_naviguation. I use the google_speech_recognition API to send a goal in Rviz for the robot to go to. The demos works well. 

I also modified the structure of the si_utils so we can import the .py in other codes.  I modified the setup_remote_pc so that the new path is taken into account.

 I added cartesian_control and movo_class. Cartesian control is a simple node that lets you use MoveIt and send end effector positions. The Movo_class lets you control the arm by sending joint positions or end effector positions. 

I also modified some READMEs to explain the demos.

### CODE REVIEW

**Comment by alexvannobel** [comment]
Does the google_speech_recognition require an Internet connection to work?

**Comment by NaderMansouri** [comment]
Yes, the Google_speech_recognition is an online speech recognition module. Movo needs to be connected to the Internet. I usually connect Movo by USB tethering to my phone so I can share my connection with it.

File: docs/Concepts/c_using_demos.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+In ROS, there is a small API, but everything is customizable with ROS. MoveIt offers certain capabilities to Move the arms and you need to adapt it for the Movo yourself.
+
+Here is a little document explaining what some of the files can do, you can always explore what the classes do:
+
+
+movo_common/movo_ros/src/movo_action_clients:
+	move_base_action_client.py is an interface to drive Movo with a certain speed during a certain time.
+		_motion_vel: You can send velocity commands to the base. For example, if you send [v_x,v_y, v_z]=[1,2,3], you are sending a command of 1 m/s in x, a command of 2 m/s in y, and a rotation command of 3degrees/second.
+		_motion_dist: Instead of sending velocity commands, you can send distance commands. You tell the base to move in x, y, z. Basically, you are sending them to a relative position from its current positions. If you want, you can record the initial position and move the base in regard to the initial position.
```
Comment (alexvannobel) [inline docs/Concepts/c_using_demos.md]: Isn't Z a rotation?

File: docs/Concepts/c_using_demos.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		add_point: This method gives a linear position for the torso and a time point for when it will be.
+	gripper_action_client.py is an interface to open and close the grippers.
+		command: Commands a position for the grippers. Position 0 corresponds to open position and position 0.165 corresponds to closed position.
+
+
+movo_common/si_utils/src/si_utils/: This folder contains example codes on how to use these classes for your needs.
+
+-------------------------------------------------------------------------------------------------------
+movo_demos:
+	XXX = robot [When you connect to the robot, ROS MASTER is Movo2] and sim [In simulation, ROS_MASTER is your computer]
```
Comment (alexvannobel) [inline docs/Concepts/c_using_demos.md]: I didn't understand the XXX thing at first glance, could you rephrase it maybe?

File: docs/Concepts/c_using_demos.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	XXX_robot_demo.launch: Launch a simple demo to showcase the robot capabilities.
+	XXX_teleop.launch: Launch the joystick teleop node. You will be able to move the robot around with the joystick connected to the computer or directly with the Movo.
+	XXX_assisted_teleop.launch: Launches the teleop.launch node and uses a footprint area around the robot to avoid collision with surrounding objects.
+	dance_invitation.launch: Movo will track your face and when it recognizes a face it will position itself and ask the user if it wants to dance. If it does, it will extend its right arm. You will 		be able to control Movo by applying forces on the end effector and make it move.
+	face_tracking: The face_tracking will launch and will follow a person's face (after recognizing it). If you move to fast, the head may not follow you.
```
Comment (alexvannobel) [inline docs/Concepts/c_using_demos.md]: "will follow a person's face" maybe just explicitely state the Kinect will follow? The whole movo following someone would be quite creepy

File: movo_common/movo_config/movo_config.bash [hunk 4]
```diff
@@ -30,9 +30,14 @@ export RUN_FULL_SYSTEM_TELEOP=true
...
-export MOVO_JOY_IS_ATTACHED=false
-export MOVO_JOY_DEV=/dev/input/js0
+# If you are using the logitech controller, put MOVO_JOY_IS_LOGITECH to true and MOVO_JOY_IS_XBOX to false
+# If you want to use the XBox 360 wireless controller, put the opposite instead. 
+# You need to do a sync_robot and a reboot of the Movo for it to apply.
 
+export MOVO_JOY_IS_ATTACHED=true
+export MOVO_JOY_DEV=/dev/input/js0
+export MOVO_JOY_IS_XBOX=true
+export MOVO_JOY_IS_LOGITECH=false
```
Comment (alexvannobel) [inline movo_common/movo_config/movo_config.bash]: LOGITECH should be true by default I think, and traditionally the MOVO_JOY_IS_ATTACHED variable was false too

File: movo_common/movo_config/movo_config.bash [hunk 5]
```diff
@@ -60,8 +65,8 @@ export MOVO_HAS_TWO_KINOVA_ARMS=true
 export KINOVA_RIGHT_ARM_IP_ADDRESS=10.66.171.15
 export KINOVA_LEFT_ARM_IP_ADDRESS=10.66.171.16
 export KINOVA_ARM_IFACE=eth0
-export MOVO_HAS_KINOVA_ARM_7DOF=true
-export MOVO_HAS_KINOVA_ARM_6DOF=false
+export MOVO_HAS_KINOVA_ARM_7DOF=false
```
Comment (alexvannobel) [inline movo_common/movo_config/movo_config.bash]: 7 DOF should be true

File: movo_common/movo_ros/bin/movo_teleop_full_system_xbox [hunk 6]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+"""--------------------------------------------------------------------
+Copyright (c) 2017, Kinova Robotics inc.
```
Comment (alexvannobel) [inline movo_common/movo_ros/bin/movo_teleop_full_system_xbox]: 2019

File: movo_common/movo_ros/bin/movo_teleop_xbox [hunk 7]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+"""--------------------------------------------------------------------
+Copyright (c) 2017, Kinova Robotics inc.
```
Comment (alexvannobel) [inline movo_common/movo_ros/bin/movo_teleop_xbox]: 2019

File: movo_common/movo_ros/src/movo/README.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
+<h1>Movo Xbox teleoperation</h1>
+
+To use the Xbox 360 wireless controller instead of the logitech controller, install the Xbox driver:
+
+`sudo apt-get install xboxdrv`
+
+You should plug a wireless controller receiver in Movo or your computer.
```
Comment (alexvannobel) [inline movo_common/movo_ros/src/movo/README.md]: Maybe add the development computer has to be in the same ROS network

File: movo_common/movo_ros/src/movo/README.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+If you want to control the Movo with a logitech controller or with an Xbox controller, you should go in movo_common/movo_config/movo_config.bash and change 
+MOVO_JOY_IS_XBOX or MOVO_JOY_IS_LOGITECH. Put the joystick you want to true and the other to false. Do a `sync_robot -nc` and reboot Movo for the change
```
Comment (alexvannobel) [inline movo_common/movo_ros/src/movo/README.md]: Instead of the command, I would advise the user to recompile both catkin_ws on movo1 and movo2 and then reboot

File: movo_common/movo_ros/src/movo/movo_teleop.py [hunk 10]
```diff
@@ -32,7 +32,7 @@
  \brief  This module contains a class for teleoperating the movo
          platform with a joystick controller
 
- \Platform: Ubuntu 16.04 LTS / ROS Kinetic
+ \Platform: Linux/ROS Indigo
```
Comment (alexvannobel) [inline movo_common/movo_ros/src/movo/movo_teleop.py]: undo this

File: movo_common/movo_ros/src/movo/movo_teleop_full_system.py [hunk 11]
```diff
@@ -33,7 +33,7 @@
          platform DOF with a joystick controller; only works with logitech
          extreme 3d
 
- \Platform: Ubuntu 16.04 LTS / ROS Kinetic
+ \Platform: Linux/ROS Indigo
```
Comment (alexvannobel) [inline movo_common/movo_ros/src/movo/movo_teleop_full_system.py]: undo this

File: movo_common/movo_ros/src/movo/movo_teleop_full_system_xbox.py [hunk 12]
```diff
@@ -0,0 +1,457 @@
+"""--------------------------------------------------------------------
+Copyright (c) 2017, Kinova Robotics inc.
```
Comment (alexvannobel) [inline movo_common/movo_ros/src/movo/movo_teleop_full_system_xbox.py]: 2019

File: movo_common/movo_ros/src/movo/movo_teleop_full_system_xbox.py [hunk 13]
```diff
@@ -0,0 +1,457 @@
...
+ \file   movo_teleop_full_system_xbox.py
+
+ \brief  This module contains a class for teleoperating all the movo
+         platform DOF with a joystick controller; only works with Xbox controller
+
+ \Platform: Ubuntu 16.04 LTS / ROS Kinetic
+
+Edited 7/25/2016: Vivian Chu, vchu@gatech - included support for simulation
+Edited 11/07/2016: David Kent, dekent@gatech - integrated arm commands with wpi_jaco
+
```
Comment (alexvannobel) [inline movo_common/movo_ros/src/movo/movo_teleop_full_system_xbox.py]: You could add your edit here

File: movo_common/si_utils/src/si_utils/head_jtas_test.py [hunk 14]
```diff
@@ -112,7 +112,7 @@ def main():
     points = [list(current_angles), 0.0]
     for i in range(0,10):
         
-        pos = [random.uniform(-1.57,1.57),random.uniform(-1.57,1.57)]
+        pos = [0,random.uniform(-1.57,1.57)]
```
Comment (alexvannobel) [inline movo_common/si_utils/src/si_utils/head_jtas_test.py]: why absolutely 0?

File: movo_common/si_utils/src/si_utils/jaco_jtas_test.py [hunk 15]
```diff
@@ -117,7 +117,8 @@ def clear(self, arm='right'):
 def main():
     rospy.init_node('jaco_jtas_test')
     arm_laterality = 'right'
-    dof = rospy.get_param('/' + arm_laterality + '_arm_ctl/jaco_dof')
+    #dof = rospy.get_param('/' + arm_laterality + '_arm_ctl/jaco_dof')
+    dof = '6dof'
```
Comment (alexvannobel) [inline movo_common/si_utils/src/si_utils/jaco_jtas_test.py]: this should be undone

File: movo_demos/launch/voice_nav/README.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
+Voice navigation demo:
+
+The voice navigation demo uses the move_base node, a map and a microphone to command Movo to a certain position in said map. 
+
+Prior to using the demo, you should map the environment in which Movo will naviguate. Make sure the map is accurate and up to date to said location.
```
Comment (alexvannobel) [inline movo_demos/launch/voice_nav/README.md]: navigate*

File: movo_demos/launch/voice_nav/README.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+Also, when mapping with Movo, it may drift and create some artifacts. These can be smoothened using Paint or Gimp to make a better, cleaner map. Bare in mind
```
Comment (alexvannobel) [inline movo_demos/launch/voice_nav/README.md]: Bear*

File: movo_demos/launch/voice_nav/README.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+that if the map is too innacurate, the demo will not work. Also, keep in mind that Movo sees from its base only. A table must be drawn in gimp for Movo to avoid
```
Comment (alexvannobel) [inline movo_demos/launch/voice_nav/README.md]: "sees"? I would say "detects" or talk about the laser sensors specifically

File: movo_demos/scripts/cartesian_control.py [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+from math import pi
+from std_msgs.msg import String
+from sensor_msgs.msg import JointState
+from moveit_commander.conversions import pose_to_list
+from trajectory_msgs.msg import (JointTrajectoryPoint,)
+
+"""
+This class takes root in the moveit_ Move Group Python Interface. This code is useful because someone can plan trajectories and movements with only a python code. The link [https://ros-planning.github.io/moveit_tutorials/doc/move_group_python_interface/move_group_python_interface_tutorial.html].
+
+It has been adapted for MOVO, or at least, it should work on MOVO.
```
Comment (alexvannobel) [inline movo_demos/scripts/cartesian_control.py]: But does it? What does this comment mean?

File: movo_demos/scripts/cartesian_control.py [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+		pose_goal.position.z =z
+		self.group.set_pose_target(pose_goal)
+		plan = self.group.go(wait=True)
+		self.group.stop()
+		self.group.clear_pose_targets()
+		
+
+def main():
+
+	mon_robot = Movo_movement('left_arm')
```
Comment (alexvannobel) [inline movo_demos/scripts/cartesian_control.py]: ```mon_robot``` should be replaced by an English-named variable

File: movo_demos/scripts/movo_class.py [hunk 21]
```diff
@@ -0,0 +1,150 @@
...
+from si_utils.base_motion_test import BaseMotionTest
+from trajectory_msgs.msg import (JointTrajectoryPoint,)
+from si_utils.head_jtas_test import HeadJTASTest
+from si_utils.voice_test import MovoVoiceTest
+from si_utils.torso_jtas_test import TorsoJTASTest
+from si_utils.jaco_jtas_test import JacoJTASTest
+"""
+This class takes root in the moveit_ Move Group Python Interface. This code is useful because someone can plan trajectories and movements with only a python code. The link [https://ros-planning.github.io/moveit_tutorials/doc/move_group_python_interface/move_group_python_interface_tutorial.html].
+
+It has been adapted for MOVO, or at least, it should work on MOVO.
```
Comment (alexvannobel) [inline movo_demos/scripts/movo_class.py]: same comment as in last file for this comment

File: movo_demos/scripts/voice_commands.py [hunk 22]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+"""
+This code is meant to control MOVO with voice commands using a map created prior to the demo. In short, when I tell it to go somewhere, for example Nassif's office, it should go there.
```
Comment (alexvannobel) [inline movo_demos/scripts/voice_commands.py]: Is Nassif's office necessary or could we give a more generic example?

File: movo_demos/scripts/voice_nav.py [hunk 23]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+"""
+This code is meant to control MOVO with voice commands using a map created prior to the demo. In short, when I tell it to go somewhere, for example Nassif's office, it should go there.
```
Comment (alexvannobel) [inline movo_demos/scripts/voice_nav.py]: same comment for Nassif's office

File: movo_demos/scripts/voice_nav.py [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+
+from tf.transformations import euler_from_quaternion, quaternion_from_euler
+import rospy
+from geometry_msgs.msg import Pose, PoseStamped, PointStamped, PoseWithCovarianceStamped, Point, Quaternion, Twist
+import os
+from std_msgs.msg import String, Bool
+#The goals are predetermined positions in a map with a location name as key. They need to be changed to work with a specific map.
+
+goals = { "boss": [-3.86, 5.36, 0.000, 1],"friend": [-3.98, -3.54, 0.000, 1], "hallway":[-4.529,0.1,0.0,1.0], "home": [-0.42,0.1, 0.0,1.0], "lobby":[4.36,-4.96,0,1.0]} #test_zone
+#goals = { "boss": [-5.0, 2.34, 0.000, 0],"friend": [0.04, 2.18, 0.000, -1.385], "hallway":[-6.66,0.18,0.0,1.0], "home": [-2.44,-5.826, 0.0,1.385], "lobby":[-6.8,0.28,0,0]} #soussol2
```
Comment (alexvannobel) [inline movo_demos/scripts/voice_nav.py]: are all those commented goals still useful? If no, we should remove them to avoid confusion

File: movo_robot/movo_upstart/templates/job-start-movo.em [hunk 25]
```diff
@@ -118,7 +118,7 @@ fi
 
 temp=1
 cnts=0
-while [ $cnts -lt 10 ] && [ $temp -ne 0 ]; do
+while [ $cnts -lt 100 ] && [ $temp -ne 0 ]; do
```
Comment (alexvannobel) [inline movo_robot/movo_upstart/templates/job-start-movo.em]: why change this value?

File: movo_common/si_utils/src/si_utils/head_jtas_test.py (hunk 14 above)
Comment (NaderMansouri) [reply movo_common/si_utils/src/si_utils/head_jtas_test.py]: Haven't realized I modified it, apologies

File: movo_common/si_utils/src/si_utils/jaco_jtas_test.py (hunk 15 above)
Comment (NaderMansouri) [reply movo_common/si_utils/src/si_utils/jaco_jtas_test.py]: Apologies again, this was for debugging purposes

File: movo_demos/scripts/cartesian_control.py (hunk 19 above)
Comment (NaderMansouri) [reply movo_demos/scripts/cartesian_control.py]: It is to give credit to code found on MoveIt. I have taken the move_group_python interface and modified it specifically for Movo

File: movo_robot/movo_upstart/templates/job-start-movo.em (hunk 25 above)
Comment (NaderMansouri) [reply movo_robot/movo_upstart/templates/job-start-movo.em]: This value corresponds to the number of tries the robot will ping its own arms. Sometimes, I was getting errors telling me that Movo couldn't find right arm or left arm. By increasing the number of pings, it lets enough time for Movo to correctly ping its arms.

File: movo_common/movo_ros/src/movo/README.md (hunk 8 above)
Comment (NaderMansouri) [reply movo_common/movo_ros/src/movo/README.md]: Done

File: movo_common/movo_config/movo_config.bash (hunk 4 above)
Comment (NaderMansouri) [reply movo_common/movo_config/movo_config.bash]: Done

File: movo_common/movo_config/movo_config.bash (hunk 5 above)
Comment (NaderMansouri) [reply movo_common/movo_config/movo_config.bash]: Done

**Comment by alexvannobel** [comment]
I reviewed the changes and it's good to me, but you have to fix the conflicts before I can merge.
